sdk/storage_blobs/src/blob/operations/get_metadata.rs

```diff
@@ -0,0 +1,91 @@
+use crate::prelude::*;
+use azure_core::{headers::*, prelude::*, RequestId};
+use chrono::{DateTime, Utc};
+use std::convert::{TryFrom, TryInto};
+
+#[derive(Debug, Clone)]
+pub struct GetMetadataBuilder {
+    blob_client: BlobClient,
+    blob_versioning: Option<BlobVersioning>,
+    lease_id: Option<LeaseId>,
+    client_request_id: Option<ClientRequestId>,
+    timeout: Option<Timeout>,
+}
+
+impl GetMetadataBuilder {
+    pub(crate) fn new(blob_client: BlobClient) -> Self {
+        Self {
+            blob_client,
+            blob_versioning: None,
+            lease_id: None,
+            client_request_id: None,
+            timeout: None,
+        }
+    }
+
+    setters! {
+        blob_versioning: BlobVersioning => Some(blob_versioning),
+        lease_id: LeaseId => Some(lease_id),
+        timeout: Timeout => Some(timeout),
+        client_request_id: ClientRequestId => Some(client_request_id),
+    }
+
+    pub fn into_future(self) -> Response {
+        Box::pin(async move {
+            let mut url = self.blob_client.url_with_segments(None)?;
+
+            url.query_pairs_mut().append_pair("comp", "metadata");
+            self.blob_versioning.append_to_url_query(&mut url);
+            self.timeout.append_to_url_query(&mut url);
+
+            let mut request =
+                self.blob_client
+                    .prepare_request(url.as_str(), http::Method::GET, None)?;
+            request.add_optional_header(&self.client_request_id);
+            request.add_optional_header(&self.lease_id);
+
+            let response = self
+                .blob_client
+                .http_client()
+                .execute_request_check_status(&request)
+                .await?;
+
+            response.headers().try_into()
+        })
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct GetMetadataResponse {
+    pub request_id: RequestId,
+    pub etag: String,
+    pub server: String,
+    pub date: DateTime<Utc>,
+    pub metadata: Metadata,
+}
+
+impl TryFrom<&Headers> for GetMetadataResponse {
+    type Error = crate::Error;
+
+    fn try_from(headers: &Headers) -> Result<Self, Self::Error> {
+        debug!("headers == {:#?}", headers);
```

Comment:
```suggestion
```

Reply:
(removed all the other similar messages in the process)